pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -555,16 +555,19 @@ func (r *MemberRollReconciler) ensureMemberExists(ctx context.Context, ns, meshN
 }
 
 func (r *MemberRollReconciler) finalizeObject(ctx context.Context, roll *maistrav1.ServiceMeshMemberRoll) (ok bool, err error) {
-	for _, ns := range roll.Spec.Members {
-		member := maistrav1.ServiceMeshMember{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      common.MemberName,
-				Namespace: ns,
-			},
+	members := &maistrav1.ServiceMeshMemberList{}
+	err = r.Client.List(ctx, members, client.MatchingFields{"spec.controlPlaneRef.namespace": roll.Namespace})
+	if err != nil {
+		return false, err
+	}
+
+	for _, member := range members.Items {
+		if member.DeletionTimestamp != nil || !isCreatedByThisController(&member) {
```

Comment:
Should this be `== nil`?

Reply:
I do have what I call booleanitis, where I flip booleans, but in this case, `skip deletion if deletionTimestamp is set` is correct, since `deletionTimestamp` is set, the member is already deleted and doesn't need to be.

Reply:
got it, thanks